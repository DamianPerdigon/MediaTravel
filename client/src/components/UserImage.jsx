// Importing the Box component from Material-UI
import { Box } from "@mui/material";

// UserImage component for displaying user images
const UserImage = ({ image, size = "60px" }) => {
  return (
    // Box component to encapsulate the image, setting its width and height
    <Box width={size} height={size}>
      <img
        // Styling the image to be circular and cover the entire area
        style={{ objectFit: "cover", borderRadius: "50%" }}
        width={size}
        height={size}
        alt="user"
        src={`http://localhost:3001/assets/${image}`}
      />
    </Box>
  );
};

export default UserImage;
