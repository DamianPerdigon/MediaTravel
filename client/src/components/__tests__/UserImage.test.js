import React from 'react';
import { render } from '@testing-library/react';
import UserImage from '../UserImage';

describe('UserImage', () => {
  it('debe coincidir con el snapshot', () => {
    const { asFragment } = render(
      <UserImage image="test-image.jpg" size="60px" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
