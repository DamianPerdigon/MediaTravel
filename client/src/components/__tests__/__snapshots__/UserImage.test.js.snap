// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserImage debe coincidir con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-13gzsmf"
  >
    <img
      alt="user"
      height="60px"
      src="http://localhost:3001/assets/test-image.jpg"
      style="object-fit: cover; border-radius: 50%;"
      width="60px"
    />
  </div>
</DocumentFragment>
`;
